### ISSUE TITLE
Error using DBvisualiser

### ISSUE THREAD

**Issue by GregBowyer** [facebook/presto#3758]
Hi all, looks like the following query

```sql
SELECT TABLE_CAT, TABLE_SCHEM, TABLE_NAME, COLUMN_NAME, DATA_TYPE,
  TYPE_NAME, COLUMN_SIZE, BUFFER_LENGTH, DECIMAL_DIGITS, NUM_PREC_RADIX,
  NULLABLE, REMARKS, COLUMN_DEF, SQL_DATA_TYPE, SQL_DATETIME_SUB,
  CHAR_OCTET_LENGTH, ORDINAL_POSITION, IS_NULLABLE,
  SCOPE_CATALOG, SCOPE_SCHEMA, SCOPE_TABLE,
  SOURCE_DATA_TYPE, IS_AUTOINCREMENT, IS_GENERATEDCOLUMN
FROM system.jdbc.columns

WHERE TABLE_CAT = 'dwh' AND TABLE_SCHEM LIKE 'default' AND TABLE_NAME LIKE 'rate_rules' AND COLUMN_NAME LIKE '%'
ORDER BY TABLE_CAT, TABLE_SCHEM, TABLE_NAME, ORDINAL_POSITION
```

Makes the following exception
```
2015-10-14T09:15:25.305Z        ERROR   remote-task-callback-11 com.facebook.presto.execution.StageStateMachine Stage 20151014_091525_00015_8gsza.1 failed
java.lang.IllegalArgumentException: Expected value 4 to be an instance of Long, but is a Integer
        at com.facebook.presto.spi.InMemoryRecordSet.checkArgument(InMemoryRecordSet.java:256)
        at com.facebook.presto.spi.InMemoryRecordSet.access$500(InMemoryRecordSet.java:36)
        at com.facebook.presto.spi.InMemoryRecordSet$Builder.addRow(InMemoryRecordSet.java:229)
        at com.facebook.presto.connector.system.jdbc.ColumnJdbcTable.addColumnRows(ColumnJdbcTable.java:126)
        at com.facebook.presto.connector.system.jdbc.ColumnJdbcTable.cursor(ColumnJdbcTable.java:113)
        at com.facebook.presto.connector.system.SystemRecordSetProvider$1.cursor(SystemRecordSetProvider.java:116)
        at com.facebook.presto.split.MappedRecordSet.cursor(MappedRecordSet.java:53)
        at com.facebook.presto.spi.RecordPageSource.<init>(RecordPageSource.java:37)
        at com.facebook.presto.split.RecordPageSourceProvider.createPageSource(RecordPageSourceProvider.java:41)
        at com.facebook.presto.split.PageSourceManager.createPageSource(PageSourceManager.java:48)
        at com.facebook.presto.operator.ScanFilterAndProjectOperator.createSourceIfNecessary(ScanFilterAndProjectOperator.java:254)
        at com.facebook.presto.operator.ScanFilterAndProjectOperator.isFinished(ScanFilterAndProjectOperator.java:169)
        at com.facebook.presto.operator.Driver.processInternal(Driver.java:377)
        at com.facebook.presto.operator.Driver.processFor(Driver.java:303)
        at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:587)
        at com.facebook.presto.execution.TaskExecutor$PrioritizedSplitRunner.process(TaskExecutor.java:505)
        at com.facebook.presto.execution.TaskExecutor$Runner.run(TaskExecutor.java:640)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)


2015-10-14T09:15:25.305Z        ERROR   query-execution-10      com.facebook.presto.execution.QueryStateMachine Query 20151014_091525_00015_8gsza failed
java.lang.IllegalArgumentException: Expected value 4 to be an instance of Long, but is a Integer
        at com.facebook.presto.spi.InMemoryRecordSet.checkArgument(InMemoryRecordSet.java:256)
        at com.facebook.presto.spi.InMemoryRecordSet.access$500(InMemoryRecordSet.java:36)
        at com.facebook.presto.spi.InMemoryRecordSet$Builder.addRow(InMemoryRecordSet.java:229)
        at com.facebook.presto.connector.system.jdbc.ColumnJdbcTable.addColumnRows(ColumnJdbcTable.java:126)
        at com.facebook.presto.connector.system.jdbc.ColumnJdbcTable.cursor(ColumnJdbcTable.java:113)
        at com.facebook.presto.connector.system.SystemRecordSetProvider$1.cursor(SystemRecordSetProvider.java:116)
        at com.facebook.presto.split.MappedRecordSet.cursor(MappedRecordSet.java:53)
        at com.facebook.presto.spi.RecordPageSource.<init>(RecordPageSource.java:37)
        at com.facebook.presto.split.RecordPageSourceProvider.createPageSource(RecordPageSourceProvider.java:41)
        at com.facebook.presto.split.PageSourceManager.createPageSource(PageSourceManager.java:48)
        at com.facebook.presto.operator.ScanFilterAndProjectOperator.createSourceIfNecessary(ScanFilterAndProjectOperator.java:254)
        at com.facebook.presto.operator.ScanFilterAndProjectOperator.isFinished(ScanFilterAndProjectOperator.java:169)
        at com.facebook.presto.operator.Driver.processInternal(Driver.java:377)
        at com.facebook.presto.operator.Driver.processFor(Driver.java:303)
        at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:587)
        at com.facebook.presto.execution.TaskExecutor$PrioritizedSplitRunner.process(TaskExecutor.java:505)
        at com.facebook.presto.execution.TaskExecutor$Runner.run(TaskExecutor.java:640)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

Using parquet files on hdfs via hive if that helps.

**Comment by electrum** [comment]
Fixed in #3760

**Comment by GregBowyer** [comment]
Many thanks for the fast turnaround
